cmd/testscript/main.go

```diff
@@ -282,5 +290,16 @@ func run(runDir, filename string, verbose bool, envVars []string) error {
 		return fmt.Errorf("error running %v in %v\n", renderFilename(filename), runDir)
 	}
 
+	if update {
+		// Write the testscript archive back to the source file
+		source, err := ioutil.ReadFile(scriptFile)
+		if err != nil {
+			return fmt.Errorf("failed to read from script file %v for -update: %v", scriptFile, err)
+		}
+		if err := ioutil.WriteFile(filename, source, 0666); err != nil {
```

Comment:
What if we're reading from stdin and filename is `"-"` ?

Reply:
Good catch. I thought about that earlier, but then clearly got distracted.

Reply:
The solution I'm just about to push up retains the ability for the user to call `testscript` with a combination of `-` and a regular file. Whilst this seems a bit bizarre, no harm in retaining the ability IMO - thoughts?